### PR TITLE
[move] Enable runtime checks in MoveVM

### PR DESCRIPTION
**Pull request by runtian-zhou**
### Description

Update the Move commit hash to enable runtime type checks in MoveVM. This is another defense measure against possible type confusions in the move bytecode verifier.

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4547)

### CODE REVIEW

File: aptos-move/aptos-vm/src/aptos_vm.rs [hunk 1]
```diff
@@ -114,6 +117,24 @@ impl AptosVM {
         }
     }
 
+    /// Sets runtime config when invoked the first time.
+    pub fn set_runtime_config(mut runtime_config: RuntimeConfig) {
```
Comment (wrwg) [inline aptos-move/aptos-vm/src/aptos_vm.rs:121]: Unnecessary mut?

File: aptos-move/move-deps/Cargo.toml [hunk 2]
```diff
@@ -21,34 +21,34 @@ edition = "2021"
...
-move-table-extension = { git = "https://github.com/move-language/move", rev = "94552a7fd7381b84376f6d7008d1f3110b5eccc5" }
-move-transactional-test-runner = { git = "https://github.com/move-language/move", rev = "94552a7fd7381b84376f6d7008d1f3110b5eccc5" }
-move-unit-test = { git = "https://github.com/move-language/move", rev = "94552a7fd7381b84376f6d7008d1f3110b5eccc5" }
-move-vm-runtime = { git = "https://github.com/move-language/move", features = ["lazy_natives"], rev = "94552a7fd7381b84376f6d7008d1f3110b5eccc5" }
-move-vm-test-utils = { git = "https://github.com/move-language/move", rev = "94552a7fd7381b84376f6d7008d1f3110b5eccc5" }
-move-vm-types = { git = "https://github.com/move-language/move", rev = "94552a7fd7381b84376f6d7008d1f3110b5eccc5" }
-read-write-set = { git = "https://github.com/move-language/move", rev = "94552a7fd7381b84376f6d7008d1f3110b5eccc5" }
-read-write-set-dynamic = { git = "https://github.com/move-language/move", rev = "94552a7fd7381b84376f6d7008d1f3110b5eccc5" }
+move-abigen = { git = "https://github.com/move-language/move", rev = "44f637b52e2f921ae2da7130e7d5af32fed7aa4b" }
```
Comment (wrwg) [inline aptos-move/move-deps/Cargo.toml:24]: You are going to change this to the aptos branch commit once it lands, right?

File: config/src/config/execution_config.rs [hunk 3]
```diff
@@ -46,6 +48,8 @@ impl Default for ExecutionConfig {
             // Parallel execution by default.
             concurrency_level: 8,
             num_proof_reading_threads: 32,
+            paranoid_type_verification: true,
+            paranoid_hot_potato_verification: true,
```
Comment (davidiw) [inline config/src/config/execution_config.rs:52]: @sherry-x , we need to discuss the right values for this in our deployment configs before we should land this.

**Review by davidiw — DISMISSED** [review]
PR looks good to me, but we should at least circle back on deployment strategies and get some numbers

**Comment by davidiw** [comment]
Less than 10% TPS hit: performance benchmark with full nodes : 6978 TPS, 5685 ms latency, 7200 ms p99 latency,(!) expired 140 out of 2980140 txns

**Comment by runtian-zhou** [comment]
This is dependent on https://github.com/move-language/move/pull/509.

**Comment by runtian-zhou** [comment]
Also @msmouse just ran the check with all txns in testnet and no false positives has been triggered!

File: aptos-move/aptos-vm/src/aptos_vm.rs [hunk 4]
```diff
@@ -1045,6 +1070,17 @@ impl VMAdapter for AptosVM {
                 let (vm_status, output) =
                     self.execute_user_transaction(data_cache, txn, log_context);
 
+                if let Err(DiscardedVMStatus::UNKNOWN_INVARIANT_VIOLATION_ERROR) =
+                    vm_status.clone().keep_or_discard()
```
Comment (wrwg) [inline aptos-move/aptos-vm/src/aptos_vm.rs:1074]: Good. We should do this way more extensively (e.g. have counters etc. for all unexpected VM violations, there are more codes than this one.) At least this is a start.

File: config/src/config/execution_config.rs (hunk 3 above)
Comment (wrwg) [reply config/src/config/execution_config.rs:52]: We go now default to be on? Good with me.

**Review by wrwg — APPROVED** [review]

**Comment by wrwg** [comment]
> ### Description
> Just a PR to test the performance of the runtime checks.
> 
> This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4547)

Maybe update this comment

File: config/src/config/execution_config.rs (hunk 3 above)
Comment (runtian-zhou) [reply config/src/config/execution_config.rs:52]: I think we will turn it on by default. We've tested it with all txns in testnet and no positives has been triggered

**Review by zekun000 — APPROVED** [review]

**Pull request merged**